libc/test/src/search/lfind_test.cpp

```diff
@@ -0,0 +1,46 @@
+//===-- Unittests for lfind -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/search/lfind.h"
+#include "test/UnitTest/Test.h"
+
+int compar(void *a, void *b) {
+  return *reinterpret_cast<int *>(a) - *reinterpret_cast<int *>(b);
```

Comment:
This could also be:
```suggestion
  return *reinterpret_cast<int *>(a) == *reinterpret_cast<int *>(b);
```
which I think is more straightforward for what the compare function is doing.

Reply:
[In C, the `==` operator returns `1` if the operands are equal, and `0` if they are not.](https://godbolt.org/z/EooeKfdjP)

But what we want is to return `0` when the operands are equal. Which is the opposite of what the `==` operator does. So, we can `!=` here.

I'm sure thats what you meant here. 

To me the subtraction is clearer, but I'll defer to your judgment and change this to use `!=` operator instead.

Reply:
This is resolved in [3e3da44](https://github.com/llvm/llvm-project/pull/114692/commits/3e3da44de8e3d7bf4380853b9447216125cb287e).

This commit also had one more change. I should have changed the signature of `compar` in `lfind_test.cpp` to have const parameters in [5b4b7ca](https://github.com/llvm/llvm-project/pull/114692/commits/5b4b7ca9afbcaf1e216cd605a54fd41f1fb4108e), but I forget and left it out. This change is required by fixing the function signature of `lfind` to match the standard exactly.

Reply:
Sorry, I mixed up whether the callback should return `0` or `1` to indicate a match. I personally think `!=` would be more readable, though it's just a slight preference.